### ISSUE TITLE
no sound at all

### ISSUE THREAD

**Issue by tomcres** [AppleWin/AppleWin#716]
This is truly bizarre. Sound was working fine before I did a clean install of Windows 10. Now I can't get any built-in sound output at all from AppleWin (i.e., the system beeper/clicker). I have literally tried every combination of sound settings imaginable, but can't even get a simple beep. I have looked at the Windows mixer and verified that the sound is not muted and is in fact at 100% for AppleWin when it is running. What is more perplexing is that the Mockingboard output works fine. If I start Skyfox, I get music!

Anyway, here is the registry dump of my current settings in case that helps, but as I said, I have tried numerous combinations of sound options (Mockingbird enabled/disabled, etc.) but I don't even get so much as the system beep on boot!

`Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\AppleWin]

[HKEY_CURRENT_USER\Software\AppleWin\CurrentVersion]

[HKEY_CURRENT_USER\Software\AppleWin\CurrentVersion\Configuration]
"Printer Filename"="D:\\AppleWin1.29.0.0\\Printer.txt"
"Version"="1.29.0.0"
"Video Emulation"="2"
"Half Scan Lines"="1"
"Monochrome Color"="12632256"
"Serial Port Name"=""
"Custom Speed"="0"
"Emulation Speed"="10"
"Joystick0 Emu Type v3"="1"
"Joystick1 Emu Type v3"="0"
"PDL X-Trim"="0"
"PDL Y-Trim"="0"
"ScrollLock Toggle"="0"
"Joystick Cursor Control"="1"
"Autofire"="0"
"Joystick Centering Control"="1"
"Mouse crosshair"="0"
"Mouse restrict to window"="0"
"Sound Emulation"="3"
"Speaker Volume"="0"
"Mockingboard Volume"="0"
"Enhance Disk Speed"="0"
"Save State On Exit"="0"
"Dump to printer"="0"
"Convert printer encoding for clones"="1"
"Filter unprintable characters"="1"
"Append to printer file"="0"
"Printer idle limit"="10"
"Apple2 Type"="16"
"CPU Type"="1"
"Slot 4"="3"
"Slot 5"="3"
"Video Style"="1"

[HKEY_CURRENT_USER\Software\AppleWin\CurrentVersion\Preferences]
"Last Harddisk Image 1"=""
"Last Harddisk Image 2"=""
"Window X-Position"="369"
"Window Y-Position"="136"
"Last Disk Image 1"="Master.dsk"
"Starting Directory"="D:\\AppleWin1.29.0.0\\"

`

**Comment by tomcw** [comment]
Can you attach your AppleWin.log file?
From a command prompt, just run `apple.exe -log` and this will create a applewin.log.
(You may need to rename to .txt to attach to this issue)

**Comment by tomcres** [comment]
Sure. Here is the log.
[AppleWin.log](https://github.com/AppleWin/AppleWin/files/3801510/AppleWin.log)

**Comment by tomcw** [comment]
I agree, very odd!

NB, these Registry entries are correct, as a value of 0 is the max volume (with 59 being minimum):
>"Speaker Volume"="0"
"Mockingboard Volume"="0"

Also the log files doesn't have any red flags.

You could try `applewin.exe -no-mb`, which will initialise the Speaker but not the Mockingboard DirectSound resources. As Mockingboard is init'ed before the Speaker, then maybe it's stealing all the resources on your PC? (But then the .log should have logged an error...)

You could also try older versions of AppleWin, eg. 1.25.0.4 [here](https://github.com/AppleWin/AppleWin/releases/download/v1.25.0.4/AppleWin1.25.0.4.zip).

**Comment by tomcres** [comment]
Interestingly, when I specify -no-mb, it immediately quits about as soon as the window appears on the screen. I get the same result with the -m switch. Here is the log from running it with -no-mb:

`*** Logging started: 11/3/2019 8:31:25 AM
CmdLine: -no-mb -log
AppleWin version: 1.29.0.0
Init: CoInitializeEx(), hr=0x00000000
Init: SysClk_InitTimer(), res=1
Init: GdiSetBatchLimit()
Init: GetProgramDirectory()
Init: RegisterExtensions()
Init: FrameRegisterClass()
Init: ImageInitialize()
Main: LoadConfiguration()
Main: DebugInitialize()
Main: JoyInitialize()
Main: VideoInitialize()
Main: FrameCreateWindow() - pre
WM_CREATE
WM_CREATE: CreateGdiObjects()
0: Primary Sound Driver - 
1: Speakers (High Definition Audio Device) - {0.0.0.00000000}.{14c3d6ad-e88f-46f4-b5a4-81d13eb95e81}
2: Digital Audio (S/PDIF) (High Definition Audio Device) - {0.0.0.00000000}.{c399d5ab-a73e-4a6b-9acd-96cbf54ea75e}
Number of sound devices = 3
DSCreate succeeded for sound device #0
WM_CREATE: DSInit()
DirectInputInit: g_bDisableDirectInput=0
DirectInputUninit
DirectInputInit: DirectInputUninit()
DirectInputInit: DirectInput8Create(), hr=0x00000000
DirectInputInit: CreateDevice(), hr=0x00000000
DirectInputInit: SetDataFormat(), hr=0x00000000
DirectInputInit: SetCooperativeLevel(), hr=0x00000000
DirectInputInit: Acquire(), hr=0x00000000
DirectInputInit: SetTimer(), id=0x00000001
WM_CREATE: DIMouse::DirectInputInit()
MB_Initialize: g_bDisableDirectSound=0, g_bDisableDirectSoundMockingboard=1
WM_CREATE: MB_Initialize()
Spkr Config: soundtype = 1 (WAVE)
WM_CREATE: SpkrInitialize()
WM_CREATE: DragAcceptFiles()
WM_CREATE (done)
FrameCreateWindow: SetTimer(), id=0x00000002
Main: FrameCreateWindow() - post`

**Comment by tomcw** [comment]
>Interestingly, when I specify -no-mb, it immediately quits 

OK, this is causing a crash (these debug switches clearly haven't been exercised in a while!).

To avoid the crash, then:

1. Don't use -m or -no-mb, and start AppleWin.
2. Under Configuration->Sound, select 'No sound cards', then OK.
3. Quit AppleWin.
4. Restart with AppleWin using -no-mb

(I'll get the crash fixed.)

**Comment by tomcres** [comment]
OK. It doesn't immediately crash now, but still I get no sound from the built-in Apple speaker device. I have been (for expediency's sake) testing this by booting the included MASTER.DSK. I am assuming that the emulated Apple IIe should beep upon booting as well as emit a beep when issued a `PRINT CHR$(7)` from AppleSoft BASIC. It does neither. Logfile is attached.
[AppleWin.log](https://github.com/AppleWin/AppleWin/files/3801654/AppleWin.log)

**Comment by tomcw** [comment]
Yes, you should get a beep when booting MASTER.DSK, and also whenever you `PRINT CHR$(7)` (which can be simplified to CTRL+G).

You log looks similar to mine.

For some reason AppleWin can't register the PrintScreen key on your system:
>Unable to register for PrintScreen key(s):
. PrintScreen

But this is unrelated to this issue. (But could be OneDrive, DropBox, SnagIt, etc)

All very strange. Currently I'm out of ideas!

**Comment by tomcw** [comment]
btw, I assume you've tested your PC speakers? The AppleWin "speaker" beep will only come out of one of the speakers (either left or right, but not both). Whereas Mockingboard sound will come out of both. Perhaps one of your PC isn't well connected?

**Comment by tomcres** [comment]
I will try a couple of other things like updating the DirectX runtime and my audio chipset drivers. Who knows, maybe the fact that it was working before I re-installed Windows might have been due to updated drivers/DirectX DLLs that I don't have yet. It is truly odd though that it would be specifically the emulated internal speaker that's not working while the Mockingbird emulation appears to work fine.

**Comment by tomcres** [comment]
Eureka! Installing the Realtek-specific audio driver supplied by ASUS results in getting the speaker working in AppleWin. Apparently, even though the Microsoft-supplied generic HD Audio driver seems to work with 100% of everything else, AppleWin doesn't like it for some reason.

**Comment by tomcw** [comment]
That's great you got it working!

So a driver issue... I'll add a note to the Help's troubleshooting section for anyone else who hits this in the future.

**Comment by tomcw** [comment]
I had exactly the same issue this weekend when setting up my wife's new Win10 laptop. Even after updating the Reaktek audio driver the problem persisted. (My experience was Speaker=silent, Mockingboard=playback was variable speed, sometimes noisy.)

In the end I noticed that AppleWin's Config->Sound "Device" defaults to **disabled**!
Setting "Device = Sound Card" fixed the issue.

And I found an old bug semi-covering this issue: #355 

I'll keep this closed, but will make sure that this old bug gets fixed!
